listener.go

```diff
@@ -191,8 +191,19 @@ func ensureFilterRule(
 	subName string,
 	filterName string,
 	filter servicebus.FilterDescriber) error {
+	rules, err := sm.ListRules(ctx, subName)
+	if err != nil {
+		return err
+	}
+	for _, rule := range rules {
+		if rule.Name == filterName {
+			// exit early cause the rule already exists
+			return nil
+		}
+	}
+
 	// remove the default rule, which is the "TrueFilter" that accepts all messages
-	err := sm.DeleteRule(ctx, subName, "$Default")
+	err = sm.DeleteRule(ctx, subName, "$Default")
```

Comment:
what happens if we don't set any rules? do we remove this? will we receive messages then?

Reply:
if we don't set any rules then the Default rule will remain as this function won't be called

Reply:
however currently there's no mechanism to retroactively go delete any rules on a subscription that already has existing rules through this library (you have to go through the azure-service-bus-go SDK or other means). You can modify existing rules, however.

Reply:
that's ok. we don't write a rule management, just a pubsub wrapper :)